src/runtime/mod.rs

```diff
@@ -75,20 +81,23 @@ impl AsyncRuntime {
     }
 
     /// Await on a future for a maximum amount of time before returning an error.
-    pub(crate) async fn await_with_timeout<F>(
+    pub(crate) async fn timeout<F: Future>(
```

Comment:
this was required for the resolver, and I figured it was a better replacement for the one I wrote.